crates/wiggle/src/lib.rs

```diff
@@ -522,17 +533,19 @@ impl<'a, T> GuestPtr<'a, [T]> {
     /// This function will return a `GuestSlice` into host memory if all checks
     /// succeed (valid utf-8, valid pointers, memory is not borrowed, etc). If
     /// any checks fail then `GuestError` will be returned.
```

Comment:
Could you update the documentation of this function to indicate what the `Option` return means?

Reply:
Or sort of more generally I think it's worth writing down somewhere the general idea behind shared memory and how it works in wiggle, and then link to that as part of various comments.